### PR TITLE
Pause, Resume, Release&Reopen, Add and Remove Logging from command line

### LINKED ISSUE #1437: Reload Log Files
If it's not currently possible, could gitea re-open log files on SIGHUP? If it is possible, could somebody tell me what I'm doing wrong? At the moment, I have logrotate restarting the gitea service.

### LINKED ISSUE #9960: [Feature & Proposal] Halting Log Output and Graceful Restart/Stop
## Description

As @zeripath mentioned in #9930, we need a way to halt log output. This may be required for using `logrotate` for example.

We can introduce graceful restart and stop to achieve the feature mentioned above. In graceful stop, the Gitea process isn't immediately stopped, but let the remaining requests to be processed prior to stop. Graceful restart use similar technique, instead that new requests are served using new *generation* of Gitea process.

### PR DESCRIPTION
**Pull request by zeripath**
Provide manager commands to:

* Release-and-reopen files and connections for logging
* Pause logging
* Resume logging 
* Add a new logger
* Remove a logger

Adds a binding to `SIGUSR1` to cause gitea to release and reopen its log files. (Please note this is not `kill -1` - that causes a graceful restart - but rather `kill -USR1` or `kill -10`.)

* Fix #9960
* Fix #1437 
* Related and will unblock #9930

### CODE REVIEW

File: modules/private/manager.go [hunk 1]
```diff
@@ -81,3 +82,92 @@ func FlushQueues(timeout time.Duration, nonBlocking bool) (int, string) {
 
 	return http.StatusOK, "Flushed"
 }
+
+// PauseLogging pauses logging
+func PauseLogging() (int, string) {
+	reqURL := setting.LocalURL + fmt.Sprintf("api/internal/manager/pause-logging")
```
Comment (techknowlogick) [inline modules/private/manager.go]:
```suggestion
	reqURL := fmt.Sprintf("%sapi/internal/manager/pause-logging", setting.LocalURL)
```

File: modules/private/manager.go [hunk 2]
```diff
@@ -81,3 +82,92 @@ func FlushQueues(timeout time.Duration, nonBlocking bool) (int, string) {
...
+	if resp.StatusCode != http.StatusOK {
+		return resp.StatusCode, decodeJSONError(resp).Err
+	}
+
+	return http.StatusOK, "Logging Paused"
+}
+
+// ResumeLogging resumes logging
+func ResumeLogging() (int, string) {
+	reqURL := setting.LocalURL + fmt.Sprintf("api/internal/manager/resume-logging")
```
Comment (techknowlogick) [inline modules/private/manager.go]:
```suggestion
	reqURL := fmt.Sprintf("%sapi/internal/manager/resume-logging", setting.LocalURL)
```

File: modules/private/manager.go (hunk 1 above)
Comment (zeripath) [reply modules/private/manager.go]:
```suggestion
	reqURL := setting.LocalURL + "api/internal/manager/pause-logging"
```

File: modules/private/manager.go (hunk 2 above)
Comment (zeripath) [reply modules/private/manager.go]:
```suggestion
	reqURL := setting.LocalURL + "api/internal/manager/resume-logging"
```

File: cmd/manager.go [hunk 3]
```diff
@@ -46,15 +48,208 @@ var (
...
+								Usage: "Maximum number of daily logs to keep",
+							}, cli.BoolTFlag{
+								Name:  "compress, z",
+								Usage: "Compress rotated logs",
+							}, cli.IntFlag{
+								Name:  "compression-level, Z",
+								Usage: "Compression level to use",
+							},
+						}...),
+						Action: runAddConsoleLogger,
```
Comment (zeripath) [inline cmd/manager.go]:
```suggestion
						Action: runAddFileLogger,
```

**Review by techknowlogick — APPROVED** [review]
👏 Code is great and when going through the functionality on a local build it works as expected (and documented!). Thank you!

**Review by lafriks — APPROVED** [review]

**Comment by wxiaoguang** [comment]
> * Add a new logger
> * Remove a logger

@go-gitea/maintainers 

I will remove these two commands since they seem useless.

-> refactor: private endpoints - #38964

**Pull request merged**